langkit/support/langkit_support-diagnostics-output.ads

```diff
@@ -0,0 +1,37 @@
+package Langkit_Support.Diagnostics.Output is
+   type Text_Buffer is private;
+   --  This type is used as a wrapper around text, where we'll keep line
+   --  information to avoid re-processing it everytime we need it.
+
+   function Create (Text : Text_Cst_Access) return Text_Buffer;
+   --  Create a text buffer from a pointer to text
+
+   function Get_Line
+     (Self        : in out Text_Buffer;
+      Line_Number : Positive) return Text_Type;
+   --  Get line `Line_Number`
+
+   procedure Print_Diagnostic
+     (Self : Diagnostic; Buffer : in out Text_Buffer; Path : String);
+   --  Pretty-print given diagnostic. Outputs the diagnostic on stdout
+   --  in a form that is developper friendly when used in a command line
+   --  application::
+   --
+   --      foo.bar:1:9: Incorrect foolize
+   --      1 | Foo.Bar.Foolize
+   --              ^^^^^^^
+   --
+   --  TODO: All the information necessary to print a diagnostic should
+   --  eventually be self contained.
+
+private
+   package Offset_Vectors is new Ada.Containers.Vectors (Positive, Natural);
+
+   type Text_Buffer is record
+      Text_Ptr   : Text_Cst_Access;
+      Line_Spans : Offset_Vectors.Vector;
+      --  For each line of text in buffer, store the end line (or end of
+      --  buffer) character offset.
```

Comment:
Please put an empty line between the two members, so that IDEs don’t consider that the docstring applies to both.

Reply:
GS doesn't do this.

Reply:
I added a comment to `Text_Ptr` anyway!